File: POO JAVA/Aula05/src/aula05/Banco.java
package aula05;

public class Banco {
 
    public int numConta;
    protected String tipo;
    private String dono;
    private float saldo;
    private boolean status;

    public void estadoAtual(){
        System.out.println("-----------------------");
        System.out.println("Conta: "+this.getNumConta());
        System.out.println("Tipo: "+this.getTipo());
        System.out.println("Dono: "+this.getDono());
        System.out.println("Saldo: "+this.getSaldo());
        System.out.println("Status: "+this.getStatus());
    }
    
    public Banco() {
        this.setSaldo(0);
        this.setStatus(false);
    }
    
    public void abrirConta(String t){
        this.setTipo(t);
        this.setStatus(true);
        if(this.getTipo() == "CC"){
            this.setSaldo(50);
        } else if(this.getTipo() == "CP"){
                    this.setSaldo(150);
                }
    }
    public void fecharConta(){
        if(this.getSaldo() > 0){
            System.out.println("Impossível. Existe Saldo em conta.");
        } else if(this.getSaldo() < 0){
                    System.out.println("Impossível. Conta com Saldo devedor.");
                } else
                    System.out.println("Conta encerrada com Suceço.");
    }
    public void depositar(float s){
        if(this.getStatus() == true){
            setSaldo(this.getSaldo() + s);
            System.out.println("Deposito realizado na conta "+this.getNumConta()+ " do Sr. "+this.getDono());
        } else
            System.out.println("Erro. Conta esta Fechada.");
    }
    public void sacar(float s){
        if(this.getStatus() == true){
            if(this.getSaldo() >= s){
                this.setSaldo(this.getSaldo() - s);
                System.out.println("Saque realizado com sucesso na conta "+this.getNumConta());
            } else
                System.out.println("Erro. Saldo em conta Insuficiente.");
        } else
            System.out.println("Impossível. Conta fechada.");
    }
    public void pagarMensalidade(){
        float v = 0;
        if(this.getTipo() == "CC"){
            v = 12;
        } else if(this.getTipo() == "CP"){
                    v = 20;
                }
        if(this.getStatus() == true){
            if(this.getSaldo() >= v){
                this.setSaldo(this.getSaldo() - v);
                System.out.println("Mensalidade de R$ "+v+",00 paga com sucesso na conta "+this.getNumConta());
            } else
                System.out.println("Impossível pagar com conta fechada.");
        }
    }

    
    public int getNumConta() {
        return this.numConta;
    }
    public void setNumConta(int numConta) {
        this.numConta = numConta;
    }

    public String getTipo() {
        return this.tipo;
    }
    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getDono() {
        return this.dono;
    }
    public void setDono(String dono) {
        this.dono = dono;
    }

    public float getSaldo() {
        return this.saldo;
    }
    public void setSaldo(float saldo) {
        this.saldo = saldo;
    }

    public boolean getStatus() {
        return this.status;
    }
    public void setStatus(boolean status) {
        this.status = status;
    }
}

File: POO PHP/Aula05/Banco.php
<?php

class Banco {

    public int $numConta;
    protected String $tipo;
    private String $dono;
    private float $saldo;
    private bool $status;
    
    
    //Metodos acessores
    public function getNumConta() {
        return $this->numConta;
    }
    public function getTipo() {
        return $this->tipo;
    }
    public function getDono() {
        return $this->dono;
    }
    public function getSaldo() {
        return $this->saldo;
    }
    public function getStatus() {
        return $this->status;
    }

    public function setNumConta($numConta): void {
        $this->numConta = $numConta;
    }
    public function setTipo($tipo): void {
        $this->tipo = $tipo;
    }
    public function setDono($dono): void {
        $this->dono = $dono;
    }
    public function setSaldo($saldo): void {
        $this->saldo = $saldo;
    }
    public function setStatus($status): void {
        $this->status = $status;
    }

    //------------------------------------------
    
    //Metodos construtores
    public function __construct() {
        $this->saldo = 0;
        $this->status = false;
        echo "<p>Conta criada com secesso.</p>";
    }

    //------------------------------------------
    
    //Metodos do tipo Banco
    public function abrirConta($t) {
        $this->setStatus(true);
        $this->setTipo($t);
        if ($this->getTipo() == "CC"){
            $this->setSaldo(50);
            echo "<p>Conta Aberta com secesso com R$ 50,00</p>";
        } else if($this->getTipo() == "CP"){
                    $this->setSaldo(150);
                    echo "<p>Conta Aberta com secesso com R$ 150,00</p>";
                }
    }
    public function fecharConta(){
        if($this->getSaldo() > 0){
            echo "<p>Erro! Existe Saldo em conta.</p>";
        } else if ($this->getSaldo() < 0){
                    echo "<p>Erro! Conta com saldo devedor.</p>";
                } else {
                    echo "<p>Conta encerrada com sucesso.</p>";
                }
    }
    public function depositar($d){
        if($this->getStatus() == true){
            $this->setSaldo($this->getSaldo() + $d);
            echo "<p>Deposito de R$ $d na conta de ".$this->getDono()."</p>";
        } else{
            echo "<p>Erro! A conta está fechada.</p>";
        }
    }
    public function sacar($s){
        if($this->getStatus() == true){
            if($this->getSaldo() >= $s){
                $this->setSaldo($this->getSaldo() - $s);
                echo "<p>Saque realizado com sucesso.</p>";
            } else{
                echo "<p>Saldo insuficiente.</p>";
            }
        } else{
            echo "<p>Erro! Conta Fechada.</p>";
        }
    }
    public function pagarMensalidade(){
        if($this->getTipo() == "CC"){
            $m = 12;
        } else if($this->getTipo() == "CP"){
                    $m = 20;
                }
        if($this->getStatus() == true){
            $this->setSaldo($this->getSaldo() - $m);
            echo "<p>Mensalidade paga de R$ $m na conta do ".$this->getDono()."</p>";
        } else{
            echo "<p>Erro! Problema com a conta. Status Fechada.</p>";
        }
    }
}
